master/internal/trial.go

```diff
@@ -452,6 +452,21 @@ func (t *trial) maybeAllocateTask() error {
 		Debugf("starting new trial allocation")
 
 	prom.AssociateJobExperiment(t.jobID, strconv.Itoa(t.experimentID), t.config.Labels())
+
+	// TODO: The first return value is []command.LaunchWarning{command.CurrentSlotsExceeded}.
+	// How do we want to handle this? In newExperiment() the warning is part of the response back
+	// to the client.
+	_, err = t.rm.ValidateResourcePoolAvailability(
```

Comment:
can this be moved earlier in the function, so we don't 

call this ```prom.AssociateJobExperiment(t.jobID, strconv.Itoa(t.experimentID), t.config.Labels())``` or any other functions we might not want to?